examples/simple.rs

```diff
@@ -1,40 +1,67 @@
-
 extern crate rspec;
-use rspec::context::describe;
+
 use std::io;
+use std::collections::BTreeSet;
 
 pub fn main() {
     let stdout = &mut io::stdout();
     let mut formatter = rspec::formatter::Simple::new(stdout);
-    let mut runner = describe("rspec is a classic BDD testing", |ctx| {
 
-        ctx.it("can define tests", || true);
+    #[derive(Clone, Debug)]
+    struct Environment {
+        set: BTreeSet<usize>,
+        len_before: usize,
+    }
 
-        ctx.describe("rspec use results for tests results", |ctx| {
+    let environment = Environment {
+        set: BTreeSet::new(),
+        len_before: 0,
+    };
 
-            ctx.it("passed if the return is_ok()", || Ok(()) as Result<(),()>);
-
-            ctx.it("failed if the return is_err()", || Err(()) as Result<(),()>);
+    let mut runner = rspec::given("a BTreeSet", environment, |ctx| {
+        ctx.when("not having added any items", |ctx| {
+            ctx.then("it is empty", |env| {
+                assert!(env.set.is_empty())
```

Comment:
Oh, absolutely. I guess I went with `assert!` for the sake of familiarity with rustc's test crate.